### PR TITLE
APC2016 pick task with right gripper-v5

### PR DESCRIPTION
**Pull request by pazeshun**
* Add Eus model of gripper-v5 on right hand
screenshot from 2017-01-18 23 16 06
* Add firmware for gripper-v5
* Add roseus interface for baxterrgv5 (baxter with right gripper-v5)
* Add main and sample programs for baxterrgv5

~~I'll add document about how to start task.~~ Done

### CODE REVIEW

File: jsk_2016_01_baxter_apc/launch/main_rgv5.launch [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  <!-- visualization -->
+  <node name="visualize_json"
+        pkg="jsk_apc2016_common" type="visualize_pick_json.py"
+        args="$(arg json)" />
+  <!--
+  <node name="visualize_on_rviz"
+        pkg="jsk_2015_05_baxter_apc" type="visualize_on_rviz.py">
+    <param name="json" value="$(arg json)" />
+  </node>
+  -->
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/launch/main_rgv5.launch]: no more used node for apc2016

File: jsk_2016_01_baxter_apc/launch/main_rgv5.launch [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            msg_class: sensor_msgs/Image
+            savetype: ColorImage
+            fname: image.png
+          - name: /apply_mask_right_hand/output/mask
+            msg_class: sensor_msgs/Image
+            savetype: ColorImage
+            fname: mask.png
+      </rosparam>
+    </node>
+  </group>
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/launch/main_rgv5.launch]: I think this data collection is not for current image pipeline. @wkentaro What do you think?

Comment (wkentaro) [inline jsk_2016_01_baxter_apc/launch/main_rgv5.launch]: Yeah, I am for removing these lines.

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l [hunk 3]
```diff
@@ -0,0 +1,205 @@
+;; vim: set ft=lisp:
+
+(require "package://baxtereus/baxter-util.l")
+(require "package://jsk_2016_01_baxter_apc/robots/baxterrgv5.l")
+
+(unless (find-package "JSK_2016_01_BAXTER_APC")
+  (make-package "JSK_2016_01_BAXTER_APC"))
+
+(defclass jsk_2016_01_baxter_apc::baxter-robot
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l]: `baxterrgv5-robot`

File: jsk_2016_01_baxter_apc/launch/include/astra_hand_rgv5.launch [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<launch>
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/launch/include/astra_hand_rgv5.launch]: Please reuse `astra_hand.launch`.

File: jsk_2016_01_baxter_apc/euslisp/main-rgv5.l [hunk 5]
```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env roseus
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/main-rgv5.l]: Not possible to reuse main.l?

File: jsk_2016_01_baxter_apc/launch/main_rgv5.launch [hunk 6]
```diff
@@ -0,0 +1,149 @@
+<launch>
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/launch/main_rgv5.launch]: Please reuse main.launch.

File: jsk_2016_01_baxter_apc/launch/setup_for_pick_rgv5.launch [hunk 7]
```diff
@@ -0,0 +1,90 @@
+<launch>
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/launch/setup_for_pick_rgv5.launch]: Please reuse setup_for_pick.launch.

File: jsk_2016_01_baxter_apc/setup.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+from distutils.core import setup
+from catkin_pkg.python_setup import generate_distutils_setup
+
+d = generate_distutils_setup(
+    packages=['jsk_baxter_dxl_controllers'],
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/setup.py]: Why not jsk_2016_01_baxter_apc?

File: doc/jsk_2016_01_baxter_apc/apc2016_pick_trial_rgv5.rst [hunk 9]
```diff
@@ -0,0 +1,24 @@
+APC2016 Pick Task Trial on Real World with Right Gripper-v5
+=====================================
```
Comment (pazeshun) [inline doc/jsk_2016_01_baxter_apc/apc2016_pick_trial_rgv5.rst]: fixed

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l (hunk 3 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l]: I know this is strange, but I have to overwrite `baxter-robot` class in https://github.com/start-jsk/jsk_apc/pull/2063/files#diff-953a5c100179528fcd593214aa478bb4R4 with `baxterrgv5` robot model for now.
So I named `baxterrgv5` robot model `baxter-robot` and didn't use `baxterrgv5` in euslisp codes.
In my thought, we can fix this when `baxterrgv5` robot model is merged into jsk_robot

Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l]: ? I am just saying you can name the class `jsk_2016_01_baxter_apc::baxterrgv5`.

Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l]: BTW, what is rgv5 in baxterrgv5? right gripper version 5?

Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l]: > ? I am just saying you can name the class jsk_2016_01_baxter_apc::baxterrgv5.

Ok. I want to unite names, so I should change:
* Add `baxterrgv5-robot` class for `jsk_2016_01_baxter_apc::baxterrgv5-robot`'s parent which has `baxterrgv5` robot model and utils in `baxter-util.l`.
* jsk_2016_01_baxter_apc::baxter-interface -> jsk_2016_01_baxter_apc::baxterrgv5-interface
* `*baxter*` -> `*baxterrgv5*`

> BTW, what is rgv5 in baxterrgv5? right gripper version 5?

Yes, exactly.

File: doc/jsk_2016_01_baxter_apc/apc2016_pick_trial_rgv5.rst [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  # Launch nodes in recognition pipeline for pick task.
+  baxter@sheeta $ roslaunch jsk_2016_01_baxter_apc setup_for_pick_rgv5.launch
+
+  # optional: Check sanity.
+  baxter@sheeta $ rosrun jsk_2016_01_baxter_apc check_sanity_setup_for_pick
+
+  # Run task!
+  baxter@sheeta $ roslaunch jsk_2016_01_baxter_apc main_rgv5.launch json:=$(rospack find jsk_apc2016_common)/json/test_gripper_v5.json
+  # even if you pass rviz:=false to main.launch, you need to launch yes_no_button.
+  baxter@sheeta $ rosrun jsk_2016_01_baxter_apc yes_no_button
```
Comment (pazeshun) [inline doc/jsk_2016_01_baxter_apc/apc2016_pick_trial_rgv5.rst]: I'll add video

File: jsk_2016_01_baxter_apc/euslisp/main-rgv5.l (hunk 5 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/main-rgv5.l]: Unfortunately, both `apc-init` and `apc-mainloop` are different from `main.l`

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l (hunk 3 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxterrgv5.l]: fixed

File: jsk_2016_01_baxter_apc/launch/include/astra_hand_rgv5.launch (hunk 4 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/launch/include/astra_hand_rgv5.launch]: fixed

File: jsk_2016_01_baxter_apc/launch/main_rgv5.launch (hunk 6 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/launch/main_rgv5.launch]: fixed

File: jsk_2016_01_baxter_apc/launch/setup_for_pick_rgv5.launch (hunk 7 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/launch/setup_for_pick_rgv5.launch]: fixed

File: jsk_2016_01_baxter_apc/setup.py (hunk 8 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/setup.py]: fixed

File: jsk_2016_01_baxter_apc/launch/main_rgv5.launch (hunk 2 above)
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/launch/main_rgv5.launch]: fixed

**Comment by pazeshun** [comment]
Fixed except adding video

File: jsk_2016_01_baxter_apc/scripts/create_udev_rules [hunk 11]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/scripts/create_udev_rules]: Is this really needed?
Same as rosserial rules?
https://github.com/start-jsk/jsk_apc#optional

Comment (pazeshun) [inline jsk_2016_01_baxter_apc/scripts/create_udev_rules]: Yes, the same as rosserial rules.
So I should just write rules in README.
I'll fix this.

Comment (pazeshun) [inline jsk_2016_01_baxter_apc/scripts/create_udev_rules]: fixed

File: doc/jsk_2016_01_baxter_apc/apc2016_pick_trial_rgv5.rst (hunk 10 above)
Comment (pazeshun) [inline doc/jsk_2016_01_baxter_apc/apc2016_pick_trial_rgv5.rst]: Added

**Pull request merged**